extension/js/common/core/attachment.ts

```diff
@@ -136,13 +136,22 @@ export class Attachment {
       return this.treatAsValue === 'publicKey';
     }
     return (
-      this.type === 'application/pgp-keys' ||
```

Comment:
This always return `true` even with private key attachments so an additional checks is added.